ledger/src/shred.rs

```diff
@@ -538,50 +576,39 @@ impl From<ShredData> for Shred {
 
 // Get slot, index, and type from a packet with partial deserialize
 pub fn get_shred_slot_index_type(
```

Comment:
Can `get_shred_slot_index_type` just be removed and replaced with `layout::get_slot`? It looks like the only usage outside of tests is in `process_packet` which only uses the `slot` value.

Reply:
yeah, I don't like it either, but even though it only uses `slot`, it effectively also validates `index` and `shred_type` and discards packets which have invalid `index` and `shred_type`.
So it wouldn't be the same as just `layout::get_slot`.

Reply:
it seems ok to me to replace it given that the shred is going to go through another round of decode and verification. If you want to keep it to preserve the existing behavior I'm ok with that.

Reply:
leaning on keeping it for now since it does not intervene with current changes, 
and it also gathers and submits `ShredFetchStats` metrics